src/table/utils/selections-utils.js

```diff
@@ -99,40 +104,52 @@ export const getSelectedRows = ({ selectedRows, cell, evt }) => {
 const selectCell = (state, payload) => {
   const { api, rows, colIdx } = state;
   const { cell, announce, evt } = payload;
+  const isSelectMultiValues = evt.shiftKey && evt.key.includes('Arrow');
   let selectedRows = {};
 
-  if (colIdx === -1) {
-    api.begin('/qHyperCubeDef');
-  } else if (colIdx === cell.colIdx) {
-    selectedRows = { ...rows };
-  } else {
-    return state;
-  }
+  if (colIdx === -1) api.begin('/qHyperCubeDef');
+  else if (colIdx === cell.colIdx) selectedRows = { ...rows };
+  else return state;
 
   selectedRows = getSelectedRows({ selectedRows, cell, evt });
   const selectedRowsLength = Object.keys(selectedRows).length;
   handleAnnounceSelectionStatus({
     announce,
     rowsLength: selectedRowsLength,
-    isAddition: selectedRowsLength > Object.keys(rows).length,
+    isAddition: selectedRowsLength >= Object.keys(rows).length,
   });
 
   if (selectedRowsLength) {
+    !isSelectMultiValues &&
+      api.select({
+        method: 'selectHyperCubeCells',
+        params: ['/qHyperCubeDef', Object.values(selectedRows), [cell.colIdx]],
+      });
+    return { ...state, rows: selectedRows, colIdx: cell.colIdx, isSelectMultiValues };
+  }
+
+  api.cancel();
+  return { ...state, rows: selectedRows, colIdx: -1, isSelectMultiValues };
+};
+
+const selectMultiValues = (state) => {
+  const { api, rows, colIdx, isSelectMultiValues } = state;
+
+  isSelectMultiValues &&
     api.select({
       method: 'selectHyperCubeCells',
-      params: ['/qHyperCubeDef', Object.values(selectedRows), [cell.colIdx]],
+      params: ['/qHyperCubeDef', Object.values(rows), [colIdx]],
     });
-    return { ...state, rows: selectedRows, colIdx: cell.colIdx };
-  }
 
-  api.cancel();
-  return { ...state, rows: selectedRows, colIdx: -1 };
+  return { ...state, isSelectMultiValues: false };
 };
 
 export const reducer = (state, action) => {
   switch (action.type) {
     case 'select':
       return selectCell(state, action.payload);
+    case 'selectMultiValues':
+      return selectMultiValues(state);
     case 'reset':
       return state.api.isModal() ? state : { ...state, rows: {}, colIdx: -1 };
     case 'clear':
```

Comment:
why is Object.keys(state.rows).length needed?

Reply:
because we don't have to do anything if it is already cleared.

Reply:
if it is already cleared, can this 'clear' case still run?

Reply:
maybe not anymore. Before the reducer ran twice, bc of react.strict. Feel free to remove it if you test it and it works the same